torchbenchmark/__init__.py

```diff
@@ -311,6 +311,52 @@ def check_example() -> None:
         else:
             module(*example_inputs)
 
+    @base_task.run_in_worker(scoped=True)
+    @staticmethod
+    def check_device() -> None:
+        instance = globals()["model"]
+
+         # Check this BenchmarkModel has a device attribute.
+        current_device = getattr(instance, 'device', None)
+        if current_device is None:
+            raise RuntimeError('Missing device in BenchmarkModel.')
+
+        model, inputs = instance.get_module()
+        model_name = getattr(model, 'name', None)
+
+        # Check the model tensors are assigned to the expected device.
+        for t in model.parameters():
+            model_device = t.device.type
+            if model_device != current_device:
+                raise RuntimeError(f'Model {model_name} was not set to the' \
+                                   f' expected device {current_device},' \
+                                   f' found device {model_device}.')
+
+        # Check the inputs are assigned to the expected device.
+        def check_inputs(inputs):
+            if isinstance(inputs, torch.Tensor):
+                if inputs.dim() and current_device == "cuda":
+                    # Zero dim Tensors (Scalars) can be captured by CUDA
+                    # kernels and need not match device.
+                    return
+
+                inputs_device = inputs[0].device.type
```

Comment:
`inputs.device.type` I think. (The `[0]` is making a new Tensor)

Reply:
Good catch!